api/src/repositories/submission-job-queue-repository.ts

```diff
@@ -112,4 +114,132 @@ export class SubmissionJobQueueRepository extends BaseRepository {
 
     return response.rows[0].source_transform_id;
   }
+
+  /**
+   * Fetch the next available job queue record(s).
+   *
+   * @param {number} [concurrency] The number of job queue processes to select (based on how many can be processed
+   * concurrently) (integer > 0).
+   * @param {number} [attempts] The total number of times a job will be attempted until it finishes successfully
+   * (integer >= 1). This currently leverages the revision_count column, so for an attempts of N, set to N*2.
+   * @param {number} [timeout] The maximum duration a running job can take before it is considered timed out. In this
+   * case, a job that is not complete, has not reached the attempts limit, and is older than the specified timeout, will
+   * be up for re-selection for another attempt.
+   * @return {*}  {Promise<ISubmissionJobQueueRecord[]>}
+   * @memberof SubmissionJobQueueRepository
+   */
+  async getNextUnprocessedJobQueueRecords(
+    concurrency?: number,
+    attempts?: number,
+    timeout?: number
+  ): Promise<ISubmissionJobQueueRecord[]> {
+    const knex = getKnex();
+    const queryBuilder = knex
+      .queryBuilder()
+      .select()
+      .from('submission_job_queue')
+      .where('job_end_timestamp', null)
+      .where((qb1) => {
+        qb1.orWhere('job_start_timestamp', null);
+        if (timeout) {
+          qb1.orWhere(knex.raw(`job_start_timestamp < NOW() - INTERVAL '${timeout} milliseconds'`));
+        }
+      });
+
+    if (attempts) {
+      queryBuilder.andWhere('revision_count', '<', attempts);
+    }
+
+    queryBuilder.orderBy('submission_job_queue_id', 'ASC');
+
+    if (concurrency) {
+      queryBuilder.limit(concurrency);
+    }
+
+    const response = await this.connection.knex<ISubmissionJobQueueRecord>(queryBuilder);
+
+    return response.rows;
```

Comment:
`rows === undefined`

Reply:
Same as above: rows will always be an array, so accessing the 0th index is safe, it will just return undefined if there are no elements.